File: Server/banchango/src/main/java/com/sundaegukbap/banchango/ingredient/repository/IngredientRepository.java
package com.sundaegukbap.banchango.ingredient.repository;

import com.sundaegukbap.banchango.ingredient.domain.Ingredient;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface IngredientRepository extends JpaRepository<Ingredient, Long> {
}

File: Android/core/data-api/src/main/java/com/sundaegukbap/banchango/core/data/repository/api/IngredientRepository.kt
package com.sundaegukbap.banchango.core.data.repository.api

import com.sundaegukbap.banchango.ContainerIngredient
import com.sundaegukbap.banchango.Ingredient

interface IngredientRepository {
    suspend fun getIngredientContainers(): Result<List<ContainerIngredient>>
    suspend fun addIngredientContainer(containerName: String): Result<Unit>
    suspend fun getAllIngredients(): Result<List<Ingredient>>
    suspend fun getIngredientsByNameLike(name: String): Result<List<Ingredient>>
}